source/initial_temperature/adiabatic.cc

```diff
@@ -99,18 +128,41 @@ namespace aspect
                              out.thermal_conductivities[0] / (out.densities[0] * out.specific_heat[0])
                            );
 
-      // analytical solution for the thermal boundary layer from half-space cooling model
-      const double surface_cooling_temperature = age_top > 0.0 ?
-                                                 (T_surface - adiabatic_surface_temperature) *
-                                                 erfc(this->get_geometry_model().depth(position) /
-                                                      (2 * sqrt(kappa * age_top)))
-                                                 : 0.0;
-      const double bottom_heating_temperature = (age_bottom > 0.0 && this->get_adiabatic_conditions().is_initialized()) ?
-                                                (T_bottom - adiabatic_bottom_temperature + subadiabaticity)
-                                                * erfc((this->get_geometry_model().maximal_depth()
-                                                        - this->get_geometry_model().depth(position)) /
-                                                       (2 * sqrt(kappa * age_bottom)))
-                                                : 0.0;
+      double surface_cooling_temperature = 0;
+      double bottom_heating_temperature = 0;
+      if (cooling_model == "half-space cooling")
+        {
+          // analytical solution for the thermal boundary layer from half-space cooling model
+          surface_cooling_temperature = age_top > 0.0 ?
+                                        (T_surface - adiabatic_surface_temperature) *
+                                        erfc(this->get_geometry_model().depth(position) /
+                                             (2 * sqrt(kappa * age_top)))
+                                        : 0.0;
+          bottom_heating_temperature = (age_bottom > 0.0 && this->get_adiabatic_conditions().is_initialized()) ?
+                                       (T_bottom - adiabatic_bottom_temperature + subadiabaticity)
+                                       * erfc((this->get_geometry_model().maximal_depth()
+                                               - this->get_geometry_model().depth(position)) /
+                                              (2 * sqrt(kappa * age_bottom)))
+                                       : 0.0;
+        }
+
+      if (cooling_model == "plate cooling")
+        {
+          if (depth > lithosphere_thickness)
+            {
+              surface_cooling_temperature = 0;
+            }
+          else
```

Comment:
You can make this much shorter if you change it to 
```suggestion
          surface_cooling_temperature = 0;
          if (depth <= lithosphere_thickness)
```